ledger/participant-integration-api/src/main/scala/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1027,13 +1035,11 @@ private[platform] object JdbcLedgerDao {
         |truncate table party_entries cascade;
       """.stripMargin
 
-    override protected[JdbcLedgerDao] def withAsyncCommit(block: Connection => PersistenceResponse)(
-        provided: Connection): PersistenceResponse = {
+    override protected[JdbcLedgerDao] def enableAsyncCommit(implicit conn: Connection): Unit = {
       var statement: PreparedStatement = null
       try {
-        statement = provided.prepareStatement("SET LOCAL synchronous_commit = 'off'")
+        statement = conn.prepareStatement("SET LOCAL synchronous_commit = 'off'")
         statement.execute()
-        block(provided)
       } finally {
         if (statement != null) statement.close()
       }
```

Comment:
So you don't need to run the `block` within the statement?

In that case, can you just run `conn.createStatement().executeUpdate("SET LOCAL synchronous_commit = 'off'")`?

You don't need to `close` it, because you never get a `ResultSet`, and statements don't need to be closed unless you want a fast way of closing multiple result sets.

Reply:
The block is not needed inside, the previous construct was just for means of better readability at the place of application.

Reply:
Fair enough. Then I would go for something like:

```scala
    val statement = conn.createStatement()
    try {
      statement.executeUpdate("SET LOCAL synchronous_commit = 'off'")
    } finally {
      statement.close()
    }
```

Terser, and you don't need the `null` check.

Reply:
Only if the `createStatement` method protoype would be so nice, but it can throw.

Reply:
If it throws, there's nothing to `close`. `statement` will be `null`.

Reply:
Ok, right, because the rhs throws and the assignment doesn't happen.

Reply:
Updated, but used `PreparedStatement` instead